### PR TITLE
Added/expanded Tesla/Env/Adapter/Middleware moduledocs

### PR DESCRIPTION
**Pull request by amatalai**
Feedback more than welcome (especially typos/grammar fixes) :)

### CODE REVIEW

**Comment by amatalai** [comment]
Ok, I'll probably add a few more things in the meantime

**Comment by amatalai** [comment]
What do you think about replacing normal tests by doctests for helper functions (added one commit with example)?

Advantages for me:
- additional examples for hexdocs
- guarantee that examples are correct

File: lib/tesla.ex [hunk 1]
```diff
@@ -55,13 +74,121 @@ defmodule Tesla.Client do
...
               Tesla.Env.result()
 end
 
 defmodule Tesla.Adapter do
+  @moduledoc """
+  The adapter specification
+
+  Adapter is a module that denormalize request data stored in `Tesla.Env` in order to make
+  request with lower level http client (e.g. `:httpc` or `:hackney`) and normalize response data
+  in order to store it back to `Tesla.Env`. It have to export `call/2` function.
```
Comment (teamon) [inline lib/tesla.ex]: `s/have/has`

Comment (amatalai) [reply lib/tesla.ex]: done

**Comment by amatalai** [comment]
💚 I guess next batch will include moduledoc for Tesla.Client and more docs for helpers

**Pull request merged**